packages/client/src/clients/in-app-message/message-client.ts

```diff
@@ -0,0 +1,109 @@
+import Knock from "../../knock";
+import { NetworkStatus, isRequestInFlight } from "../../networkStatus";
+
+import { InAppChannelClient } from "./channel-client";
+import {
+  FetchInAppMessagesOptions,
+  InAppMessageClientOptions,
+  InAppMessagesQueryInfo,
+} from "./types";
+
+/**
+ * Manages realtime connection to in app messages service.
+ */
+export class InAppMessageClient {
+  private knock: Knock;
+
+  constructor(
+    readonly channelClient: InAppChannelClient,
+    readonly messageType: string,
+    readonly defaultOptions: InAppMessageClientOptions = {},
+  ) {
+    this.knock = channelClient.knock;
+  }
+
+  async fetch(options: FetchInAppMessagesOptions = {}) {
+    // TODO: Create better abstraction for reading from / writing to store
+    const queryParams = {
+      ...this.defaultOptions,
+      ...options,
+      // Unset options that should not be sent to the API
+      __loadingType: undefined,
+      __fetchSource: undefined,
+    };
+
+    const queryKey = `${this.messageType}-${JSON.stringify(queryParams)}`;
```

Comment:
I'm wondering if we can just use an encoded url path + query params for a query key, vs this one-off format? For example, "/v1/users/thomas/in-app-messages/channel-id/banner?foo=bar". 

Maybe we are okay with the proposed format, but at least the query url path will include a user id, which means the cache is scoped to an individual recipient. Could also be a TODO thing.

Reply:
Yeah, I like that for the query key so that it's more unique and probably easier to work with if we need to extract the values from it! I'll make a note to update it

Reply:
Yep!

Reply:
Actually just saw that we already have on in the next PR 😄